storage-xray-udp/src/main/java/zipkin2/storage/xray_udp/UDPMessageEncoder.java

```diff
@@ -51,11 +51,13 @@
         .append(span.traceId(), 0, 8)
         .append('-')
         .append(span.traceId(), 8, 32).toString());
-    if (span.parentId() != null) writer.name("parent_id").value(span.parentId());
     writer.name("id").value(span.id());
     if (span.kind() == null
         || span.kind() != Span.Kind.SERVER && span.kind() != Span.Kind.CONSUMER) {
```

Comment:
@adriancole I am not sure what `span.kind()` has to affect whether the trace is `subsegment` or not. I think it is we should only check for `parentId`. i.e move the block on line 57 outside this block.

Reply:
In brave all spans have a parent except for the root. A span of type Server or Consumer would be a Segment in X-Ray terms because these are the initial span in a service